x/ugov/keeper/query_server.go

```diff
@@ -49,3 +51,29 @@ func (q Querier) InflationCycleEnd(ctx context.Context, _ *ugov.QueryInflationCy
 	cycleEndTime := q.Keeper(&sdkCtx).InflationCycleEnd()
 	return &ugov.QueryInflationCycleEndResponse{End: &cycleEndTime}, nil
 }
+
+// TokenBalances implements ugov.QueryServer.
+func (q Querier) TokenBalances(ctx context.Context, req *ugov.QueryTokenBalances) (*ugov.QueryTokenBalancesResponse,
+	error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	if req.Height != 0 {
```

Comment:
We need to figure out why data is not loading from requested height , it is working with cli query but not with grpc-web api